### PR TITLE
authentication improved

### PR DESCRIPTION
**Pull request by isumygin-sc**
Both username and auth token settings are dropped from VS Code settings, as it was [confusing](https://github.com/playcanvas/vscode-extension/issues/15#issuecomment-2167855926) for customers.
I introduced new API in playcanvas to retrieve user id based on auth cookie or access token, it's used to retrieve userID now.

Access token is requested when user adding a project and saved into secure storage. If token expired or removed from a project, any API call will return unauthorised and token is cleared from secure storage and re-requested from user on any  next api call.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,9 +7,9 @@ Extension that integrates with the PlayCanvas platform and helps to use a rich e
 ## Usage
 
 * Download and install extension in VS Code.
-* Generate an access token on your [PlayCanvas account page](https://playcanvas.com/account) - check [this document](https://developer.playcanvas.com/en/user-manual/api/#authorization) for details.
-* Open VS Code, enter the token and PlayCanvas username on extension settings page. Search for 'playcanvas' in VS Code settings.
-* Select "PlayCanvas: Add Project" command in VS Code command menu
+* Generate an access token on your [PlayCanvas account page](https://playcanvas.com/account) - check [this document](https://developer.playcanvas.com/en/user-manual/api/#authorization) for details, copy it to clipboard
+* Select "PlayCanvas: Add Project" command in VS Code command menu, paste your token into a message box
```
Comment (willeastcott) [inline README.md:11]:
```suggestion
* Select "PlayCanvas: Add Project" command in VS Code command menu and paste your token into the message box
```

File: README.md [hunk 2]
```diff
@@ -7,9 +7,9 @@ Extension that integrates with the PlayCanvas platform and helps to use a rich e
...
+* Work with your project in VS Code workspace
```
Comment (willeastcott) [inline README.md:12]:
```suggestion
* Work with your project in VS Code workspace.
```

File: README.md [hunk 3]
```diff
@@ -48,18 +48,18 @@ The extension supports collaboration features of PlayCanvas. Multiple users can
 
 #### Settings
 
-The extension has 3 settings - Access Token, PlayCanvas username and usePlaycanvasTypes to add types support. 
+The extension has just 1 setting - usePlaycanvasTypes to add types support. Access Token is requested when you are adding a project.
```
Comment (willeastcott) [inline README.md:51]:
```suggestion
The extension has just 1 setting: `usePlaycanvasTypes` (to add types support). An Access Token is requested when you are adding a project.
```

File: README.md [hunk 4]
```diff
@@ -48,18 +48,18 @@ The extension supports collaboration features of PlayCanvas. Multiple users can
...
 
 * An existing PlayCanvas account with an access token generated.
 
 ## Extension Settings
 
-* `playcanvas.accessToken`: Generate an access token on your [account page](https://playcanvas.com/account).
-* `playcanvas.username`: Set to your PlayCanvas username.
 * `playcanvas.usePlaycanvasTypes`: Automatically adds a reference to PlayCanvas types files for code suggestions. Line is not saved. Default is true.
 
+Playcanvas Access Token is requested when you add a project - generate an access token on your [account page](https://playcanvas.com/account).
```
Comment (willeastcott) [inline README.md:61]:
```suggestion
A PlayCanvas Access Token is requested when you add a project. Generate an access token on your [account page](https://playcanvas.com/account).
```

File: src/cloudStorageProvider.js [hunk 5]
```diff
@@ -385,24 +385,27 @@ class CloudStorageProvider {
...
-            this.projects = [];
+            projects = [];
         } else {
-            this.projects.forEach(p => { delete p.files; delete p.branches; delete p.branchId } );
+            projects.forEach(p => { delete p.files; delete p.branches; delete p.branchId } );
         }
     }
 
     refreshUri(uri) {
+        console.log('refreshUri ' + uri.path);
```
Comment (willeastcott) [inline src/cloudStorageProvider.js:395]: Just checking that you intended to leave these `console.log`s in the PR...

Comment (isumygin-sc) [reply src/cloudStorageProvider.js:398]: this is for debugging, other extensions usually log stuff. It could help us to debug customer issues.

**Pull request merged**